syntax = "proto3";

package NFMsg; 

import "NFDefine.proto";
import "NFMsgBase.proto";


//除去基础对象身上的属性外，这里全部游戏中的逻辑协议
message ReqEnterGameServer
{
	Ident 		id = 1;
	bytes 		account = 2;
    int32 		game_id = 3;
	bytes 		name = 4;
}

message ReqAckEnterGameSuccess
{
    int32 		arg = 1;
}

message ReqHeartBeat
{
    int32 		arg = 1;
}

message ReqLeaveGameServer
{
    int32 		arg = 1;
}

message PlayerEntryInfo//对象出现基本信息
{
	Ident	object_guid = 1;
	float	x = 2;
	float	y = 3;
	float	z = 4;
	int32	career_type = 5;
	int32	player_state = 6;
	bytes  config_id = 7;
	int32  scene_id = 8;
	bytes  class_id = 9;
}

message AckPlayerEntryList//对象出现列表
{
	repeated PlayerEntryInfo object_list = 1;
}

message AckPlayerLeaveList//对象离去列表
{
	repeated Ident 	object_list = 1;
}

////////////////////////////////////////////////////////////////////////////////////////////////////

message PosSyncUnit
{	
	enum EMoveType
	{
		EMT_WALK = 0;
		EET_SPEEDY = 1;
		EET_TELEPORT = 2;
	}

	Ident 	mover = 1;
    Vector3 pos = 2;
    Vector3 orientation = 3;
    int32 status = 4;
	EMoveType type = 5;
}

message ReqAckPlayerPosSync
{
    int32 sequence = 1;
	repeated PosSyncUnit sync_unit = 2;
}

////////////////////////////

message EffectData
{
	enum EResultType
	{
		EET_FAIL = 0;
		EET_SUCCESS = 1;
		EET_REFUSE = 2;
		EET_MISS = 3;	
		EET_CRIT = 4;	
		EET_ULTI = 5;	
	}
	Ident 			effect_ident = 1;
	int32  		effect_value = 2;
	EResultType  	effect_rlt = 3;
}

message ReqAckUseSkill
{
	Ident 	user = 1;
	bytes skill_id = 2;
	int32  use_index = 3;//因为客户端要先展示
	repeated EffectData 	effect_data = 4;
}

message ReqAckSwapScene
{
	int32 		transfer_type = 1;
	int32		scene_id = 2;
	int32		line_id = 3;
	float 		x = 4;
	float 		y = 5;
	float 		z = 6;
	bytes 	data = 7;
}

/////////////////////////////////////////
message ReqAckPlayerChat
{
	enum EGameChatChannel
	{
		EGCC_GLOBAL = 0;
		EGCC_CLAN = 1;
		EGCC_ROOM = 2;
		EGCC_TEAM = 3;
	}
	enum EGameChatType
	{
		EGCT_TEXT= 0;
		EGCT_VOICE = 1;
		EGCT_EMOJI = 2;
	}

	Ident	player_id = 1;
	bytes 	player_name = 2;
	EGameChatChannel  chat_channel = 3;
	EGameChatType  chat_type = 4;
	bytes chat_info = 5;
}
